shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/context/CommonBeans.java

```diff
@@ -149,16 +151,22 @@ MetaStoreEventAggregator eventAggregator() {
     return new DefaultMetaStoreEventAggregator();
   }
 
+  @Bean
+  TableSelector tableSelector(TargetReplication targetReplication) {
+    return new TableSelector(targetReplication);
+  }
+
   @Bean
   MessageReaderAdapter messageReaderAdapter(
       HiveConf replicaHiveConf,
       ApiarySqsMessageDeserializer sqsMessageSerDe,
       EventReceiverConfiguration messageReaderConfig,
-      SourceCatalog sourceCatalog) {
+      SourceCatalog sourceCatalog,
+      TableSelector tableSelector) {
     MessageReaderFactory messaReaderFactory = MessageReaderFactory
         .newInstance(messageReaderConfig.getMessageReaderFactoryClass());
     MessageReader messageReader = messaReaderFactory.newInstance(replicaHiveConf, sqsMessageSerDe);
-    return new MessageReaderAdapter(messageReader, sourceCatalog.getHiveMetastoreUris());
+    return new MessageReaderAdapter(messageReader, sourceCatalog.getHiveMetastoreUris(), tableSelector);
```

Comment:
I think if possible the Adapter should not know about the filtering of tables it should get ListenerEvents that are valid for processing. Perhaps  we need to look at a `FilteringMessageReader` that filters out the stream of event. Can be a decarator on whaterever MessageReader implementation we use currently?

Reply:
I like the idea of using a decorator to filter out the tables we're not interested in.